dask_cuda/tests/test_explicit_comms.py

```diff
@@ -250,7 +239,7 @@ def check_shuffle():
             ):
                 dask.config.refresh()  # Trigger re-read of the environment variables
                 with pytest.raises(ValueError, match="explicit-comms-batchsize"):
-                    ddf.shuffle(on="key", npartitions=4)
+                    ddf.shuffle(on="key", npartitions=4).dask
```

Comment:
I'm seeing some very strange behavior locally - I'm unable to get this to raise with the env-variable approach :/

Reply:
Works fine using `with dask.config.set(...)` (as long as there is no `dask.config.refresh()` call).

Reply:
It took me a couple of hours of debugging this too, but I finally understood why this happens @rjzamora. When `dask.config.refresh()` is called `dataframe.shuffle.method` is overwritten to `None`, and [dask-expr then calls `dask.utils.get_default_shuffle_method`](https://github.com/dask/dask-expr/blob/d7577a2ff22d47b36a45a87dd5ff2857af81341f/dask_expr/_collection.py#L920), that will [return `"p2p"`](https://github.com/dask/dask/blob/2c0ac83fce0e46a15d070b6e7a8f8fe897c00a28/dask/utils.py#L2251) because `dataframe.shuffle.method is None` now. We also [patch several of Dask's `get_default_shuffle_method`](https://github.com/rapidsai/dask-cuda/blob/72b364bd47b2fcbf31b42104dc900e746af9a47e/dask_cuda/__init__.py#L37-L45), but not `dask.utils.get_default_shuffle_method`, however, even if we patch it that doesn't work and I think this has to do with the import orders, if Dask-CUDA is loaded and patches those _after_ dask-expr imports those it will not see the patched functions and thus will default to Dask's original, which makes this approach not entirely reliable.

For this specific test you can just add `"DASK_DATAFRAME__SHUFFLE__METHOD": "tasks"` to the patched dictionary in line 249. However, after this long debugging session my conclusion is that relying on `dask.config.refresh()` can get very flaky real quick and I think the best course of action here is to check environment variables in a separate test where we launch the process already with the patched dictionary instead and here move to `with dask.config.set(...)`.

Reply:
Okay - Thanks for your help debugging this @pentschev !

I think I have an idea how to avoid the need for any `get_default_shuffle_method` patching (fingers crossed).

Reply:
Okay, yeah. It looks like we don't need to patch `get_default_shuffle_method` if we patch `Shuffle._lower`. As far as I can tell, this is the central place where `get_default_shuffle_method` is typically used to set the shuffle method to `"p2p"` anyway.